test/test-readability.js

```diff
@@ -47,11 +48,44 @@ var testPages = fs.readdirSync(testPageRoot).map(function(dir) {
   };
 });
 
+function suite(result, expectedContent, expectedMetadata) {
+  it("should return a result object", function() {
+    expect(result).to.include.keys("content", "title", "excerpt", "byline");
+  });
+
+  it("should extract expected content with jsdom", function() {
+    expect(expectedContent).eql(prettyPrint(result.content));
+  });
+
+  it("should extract expected title", function() {
+    expect(expectedMetadata.title).eql(result.title);
+  });
+
+  it("should extract expected byline", function() {
+    expect(expectedMetadata.byline).eql(result.byline);
+  });
+
+  it("should extract expected excerpt", function() {
+    expect(expectedMetadata.excerpt).eql(result.excerpt);
+  });
+}
+
+function removeCommentNodesRecursively(node) {
+  for (var n = 0, length = node.childNodes.length; n < length; n++) {
+    var child = node.childNodes[n];
+    if (!child) continue; // JSDOM is dumb sometimes :/
```

Comment:
... wait... isn't it just a bug due to removing elements in a for-loop?
You cache the length, start the loop, remove elements. If one element got removed, at the end of the loop, `node.childNodes[n]` can only return `undefined`, because the `node.childNodes` isn't of the same length that it was at the beginning of the loop any longer. Possibly, you're also forgetting to remove some comments, by skipping one element after one was removed.

You can clone the list via Array.prototype.slice.call (or Array.from) and iterate over that (with forEach, Yay \o/)